extras/indexing/src/main/java/org/apache/rya/indexing/entity/query/EntityQueryNode.java

```diff
@@ -204,35 +241,116 @@ private static void verifyAllPredicatesPartOfType(Type type, Collection<Statemen
+                        //the statement pattern has this property, check to see if the values match(if constant)
+                        final Var object = objectVariables.get(key);
+                        final RyaType entityType = entityProps.get(key).getValue();
+                        final Value entityVal = ValueFactoryImpl.getInstance().createLiteral(entityType.getData(), entityType.getDataType());
+                        if(object.isConstant()) {
+                            final Value objectVal = object.getValue();
+                            if(objectVal.equals(entityVal)) {
+                                //add to binding set
+                                resultSet.addBinding(object.getName(), entityVal);
+                            }
+                        } else {
+                            resultSet.addBinding(object.getName(), entityVal);
+                        }
+                    } else {
+                        //entity doesn't hold to statement patterns
+                    }
+                }
+            } else {
+                return new EmptyIteration<BindingSet, QueryEvaluationException>();
+            }
+        } catch (final EntityStorageException e) {
+            throw new QueryEvaluationException("Failed to evaluate the binding set", e);
+        }
+        // x = alice's SSN
+        // y = blue  <-- how do i know this is for urn:eye property? FROM THE STATEMENT PATTERN. look for the ?y in the SP, and that has the property name in it.
+        bindingSet.forEach(new Consumer<Binding>() {
+            @Override
+            public void accept(final Binding binding) {
+                resultSet.addBinding(binding);
+            }
+        });
+        final List<BindingSet> list = new ArrayList<>();
+        list.add(resultSet);
+        return new CollectionIteration<>(list);
+    }
 
-        // RETURN AN EMPTY ITERATION IF IT CAN NOT FILL IT IN!
+    /**
+     * @return - The {@link StatementPattern}s that make up this {@link EntityQueryNode}
+     */
+    public Collection<StatementPattern> getPatterns() {
+        return patterns;
+    }
 
-        // for all variables in the OBJECT portion of the SPs, fill 'em in using the entity that is stored in the index.
+    @Override
+    public int hashCode() {
+        int hashcode =
+            (subjectIsConstant ? 1 : 0) +
+            type.hashCode() +
+            subjectVar.hashCode() +
+            subjectConstant.hashCode();
+        for(final StatementPattern sp : getPatterns()) {
+            hashcode += sp.hashCode();
+        }
+        return hashcode;
+    }
 
-        // x = alice's SSN
-        // y = blue  <-- how do i know this is for urn:eye property? FROM THE STATEMENT PATTERN. look for the ?y in the SP, and that has the property name in it.
+    @Override
+    public boolean equals(final Object other) {
+        if(other instanceof EntityQueryNode) {
+            return hashCode() == other.hashCode();
```

Comment:
This isn't a valid implementation of equals. You can have non-equal things have the same hash via a hash collision.